lesson6/README.md

```diff
@@ -0,0 +1,147 @@
+# Lesson 6 - APIs and HTTP Requests
+Today we'll be learning about APIs and making HTTP requests to them!
+
+## What's an API anyway?
+You've probably heard the term thrown around all the time and might have no idea what it means - no worries! An API stands for an application program interface. Think of it as a way for one application to easily communicate with another. For example, maybe you have a blog and want it to show your Twitter feed on the side. In that case, you'd need to use Twitter's API to provide a way for your blog to access Twitter's data. This example is specific to web APIs, but you'll be using a lot of Android APIs in order to make an app (e.g., an API for your app to communicate with the camera). The Android platform has a framework API that apps use to interact with the underlying Android system.
+
+## Android APIs
+Android has a *ton* of APIs built in; they're useful for using hardware sensors, accessing storage, handling user input, and setting config info.. However, you'll have to be careful with which version of Android you're using as APIs will differ depending on that. The APIs are for using hardware sensors, accessing storage, handling user input, and setting config info.
+
+Check the sidebar in [this link](https://developer.android.com/about/versions/marshmallow/android-6.0.html) out to look at the different APIs for versions 4.1 to 6.0. [This link](https://developer.android.com/reference/android/package-summary.html) provides you with a detailed list of every Android-built API available. There are way too many to list so we won't go through them in detail, but if you've seen your phone do it, you'll find how to do it there.
+
+## [Amazon S3](https://aws.amazon.com/s3/)
+Amazon Web Services (AWS) provides something called Amazon Simple Storage Service, AKA Amazon S3. As the name implies, it's a cloud storage service. Now, instead of locally storing data on your phone you can use S3 and save it on the cloud! This is great for applications that need to share or store a lot of data.
```

Comment:
I would just explain the cloud

On Sun, Jul 10, 2016 at 2:22 PM Shruti Iyer <notifications@github.com>
wrote:

> In lesson6/README.md
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/12#discussion_r70186277>
> :
>
> > @@ -0,0 +1,147 @@
> > +# Lesson 6 - APIs and HTTP Requests
> > +Today we'll be learning about APIs and making HTTP requests to them!
> > +
> > +## What's an API anyway?
> > +You've probably heard the term thrown around all the time and might have no idea what it means - no worries! An API stands for an application program interface. Think of it as a way for one application to easily communicate with another. For example, maybe you have a blog and want it to show your Twitter feed on the side. In that case, you'd need to use Twitter's API to provide a way for your blog to access Twitter's data. This example is specific to web APIs, but you'll be using a lot of Android APIs in order to make an app (e.g., an API for your app to communicate with the camera). The Android platform has a framework API that apps use to interact with the underlying Android system.
> > +
> > +## Android APIs
> > +Android has a *ton* of APIs built in; they're useful for using hardware sensors, accessing storage, handling user input, and setting config info.. However, you'll have to be careful with which version of Android you're using as APIs will differ depending on that. The APIs are for using hardware sensors, accessing storage, handling user input, and setting config info.
> > +
> > +Check the sidebar in [this link](https://developer.android.com/about/versions/marshmallow/android-6.0.html) out to look at the different APIs for versions 4.1 to 6.0. [This link](https://developer.android.com/reference/android/package-summary.html) provides you with a detailed list of every Android-built API available. There are way too many to list so we won't go through them in detail, but if you've seen your phone do it, you'll find how to do it there.
> > +
> > +## [Amazon S3](https://aws.amazon.com/s3/)
> > +Amazon Web Services (AWS) provides something called Amazon Simple Storage Service, AKA Amazon S3. As the name implies, it's a cloud storage service. Now, instead of locally storing data on your phone you can use S3 and save it on the cloud! This is great for applications that need to share or store a lot of data.
>
> Instead of cloud, we could say internet
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/12/files/f25847161c2d9097ade0825db9d669712863d52c#r70186277>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIhRRQ5-OjsiLrgaKzUSX-vVAIBwd17Kks5qUThogaJpZM4JI4f7>
> .
>